easybuild/tools/filetools.py

```diff
@@ -590,6 +596,19 @@ def download_file(filename, url, path, forced=False):
 
     # use custom HTTP header
     headers = {'User-Agent': 'EasyBuild', 'Accept': '*/*'}
+
+    # permit additional or override headers via http_headers_fields option
+    # whose string value contains header-fields grammar: (see rfc7230)
+    #   header-fields   = *( header-field *(CR) LF )
+    #   header-field    = field-name ":" OWS field-value OWS
+    #   OWS             = ( optional white space )
+    # Note CR may be omitted for convenience (it is absorbed in OWS and stripped)
+    if http_header_fields is not None:
+        extraheaders = dict(hf.split(':') for hf in http_header_fields.split('\n') if hf.count(':') == 1)
```

Comment:
Is it a potential problem that values can not include `:` characters?

Because you can just split on the first `:`:

```
>>> key, val = 'foo : bar:baz'.split(':', 1)
>>> key
'foo '
>>> val
' bar:baz'
```